int16 speed
int16 duration
---
string result